addons/web/controllers/main.py

```diff
@@ -961,7 +962,7 @@ def resequence(self, model, ids, field='sequence', offset=0):
                            starting the resequencing from an arbitrary number,
                            defaults to ``0``
         """
-        m = request.env[model]
+        m = request.env[model] if isinstance(model, str) else model
```

Comment:
why adding a model parameter as we can retrieve it already using request.env[model] ?

Reply:
I'm not sure I understand your question.
What happens here is resequence assumes the model parameter to be a string but for the purposes of my overrides in dataset.py (_sudo_resequence) I'd like it to be able to be the model itself. That is so I can pass the model as sudo, which I can't do if the model is requested in this way.
It's the least intrusive way I found to do this.